src/runtime/hexagon_remote/host_malloc.cpp

```diff
@@ -43,7 +46,21 @@ struct ion_handle_data {
 #define ION_IOC_FREE _IOWR('I', 1, ion_handle_data)
 #define ION_IOC_MAP _IOWR('I', 2, ion_fd_data)
 
-ion_user_handle_t ion_alloc(int ion_fd, size_t len, size_t align, unsigned int heap_id_mask, unsigned int flags) {
+// For libion.so usage
+extern "C" {
```

Comment:
Actually, looking more closely, I think these are unused and should be removed. You are using the function pointers below only (which makes sense).